### PR TITLE
M3-5681: DBaaS Create UI refinements

### PR DESCRIPTION
**Pull request by HanaXu**
## Description
- Fix API IP errors not being surfaced
    - I went with displaying error notices as it was the simplest option; mapping the errors to the input fields is a bit more involved and maybe something we can revisit later
- Sorts DB engine dropdown by version descending

## How to test
- `yarn test DatabaseCreate`
- Try creating a database cluster and ensure that API IP errors surface

image

### CODE REVIEW

File: packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx [hunk 1]
```diff
@@ -275,15 +280,23 @@ const DatabaseCreate: React.FC<{}> = () => {
...
-          address: '',
+          address: '1',
+          error: '',
+        },
+        {
+          address: '2',
+          error: '',
+        },
+        {
+          address: '1',
```
Comment (HanaXu) [inline packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx:299]: Will change this back before merging

File: packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx [hunk 2]
```diff
@@ -185,6 +183,7 @@ const DatabaseCreate: React.FC<{}> = () => {
 
   const [type, setType] = React.useState<DatabaseType>();
   const [createError, setCreateError] = React.useState<string>();
+  const [apiIpErrors, setApiIpErrors] = React.useState<APIError[]>();
```
Comment (DevDW) [inline packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx:186]: maybe rename these to `ipErrorsFromAPI` and `setIPErrorsFromAPI` so the casing is a little more standard and readable?

**Review by DevDW — APPROVED** [review]
LGTM -- personally I am OK with the invalid IP errors being grouped and possibly revisiting that later.

**Review by tiffwong — APPROVED** [review]
Great job!

**Pull request merged**